custom_converter/worker.py

```diff
@@ -0,0 +1,107 @@
+import asyncio
+import dataclasses
+from typing import Any, Optional, Type
+
+import temporalio.converter
+from temporalio import workflow
+from temporalio.api.common.v1 import Payload
+from temporalio.client import Client
+from temporalio.converter import (
+    CompositePayloadConverter,
+    DefaultPayloadConverter,
+    EncodingPayloadConverter,
+)
+from temporalio.worker import Worker
+
+
+class GreetingInput:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+
+class GreetingOutput:
+    def __init__(self, result: str) -> None:
+        self.result = result
+
+
+@workflow.defn
+class GreetingWorkflow:
+    @workflow.run
+    async def run(self, input: GreetingInput) -> GreetingOutput:
+        return GreetingOutput(f"Hello, {input.name}")
+
+
+class GreetingEncodingPayloadConverter(EncodingPayloadConverter):
+    @property
+    def encoding(self) -> str:
+        return "my-greeting-encoding"
```

Comment:
Maybe use a "namespace" here like `text/greeting` to encourage this practice.

Reply:
Ok

Reply:
Fixed